src/py/pyodide/_package_loader.py

```diff
@@ -16,38 +16,97 @@
 TAR_TYPES = {".tar", ".gz", ".bz", ".gz", ".tgz", ".bz2", ".tbz2"}
 
 
-def unpack_buffer(filename: str, buffer: JsProxy, target: str = "site") -> JsProxy:
+def make_whlfile(*args, owner=None, group=None, **kwargs):
+    return shutil._make_zipfile(*args, **kwargs)  # type: ignore[attr-defined]
+
+
+if IN_BROWSER:
+    shutil.register_archive_format("whl", make_whlfile, description="Wheel file")
+    shutil.register_unpack_format(
+        "whl", [".whl", ".wheel"], shutil._unpack_zipfile, description="Wheel file"  # type: ignore[attr-defined]
+    )
+
+
+def get_format(format: str) -> str:
+    for (fmt, extensions, _) in shutil.get_unpack_formats():
+        if format == fmt:
+            return fmt
+        if format in extensions:
+            return fmt
+        if "." + format in extensions:
+            return fmt
+    raise ValueError(f"Unrecognized format {format}")
+
+
+def unpack_buffer(
+    buffer: JsProxy,
+    *,
+    filename: str = "",
+    format: str = None,
+    target: Literal["site", "lib", None] = None,
+    extract_dir: str = None,
+    calculate_dynlibs: bool = False,
+) -> Optional[JsProxy]:
     """Used to install a package either into sitepackages or into the standard
     library.
 
     This is a helper method called from ``loadPackage``.
 
     Parameters
     ----------
+    buffer
+        A Javascript ``Uint8Array`` with the binary data for the archive.
+
     filename
         The name of the file we are extracting. We only care about it to figure
-        out whether the buffer represents a tar file or a zip file.
+        out whether the buffer represents a tar file or a zip file. Ignored if
+        format argument is present.
 
-    buffer
-        A Javascript ``Uint8Array`` with the binary data for the archive.
+    format
+        Controls the format that we assume the archive has. Overrides the file
+        extension of filename.
```

Comment:
It's not very clear what happens when format=None. Maybe we could say it more explicitly.

Reply:
So the options here are:
1. If format is present, we use that.
2. If file name is present, it should have an extension, like `a.zip`, `a.tar`, etc. Then we use that.
3. If neither is present or the file name has no extension, `shutil.unpack_archive` throws an unknown format error.

Reply:
I imagine. I meant that option 2 (which is the default) could be documented as well.